determined-ai/determined
# chore: update to `react-router-dom` v6

## Description
[WEB-172]
[WEB-518]

Previous: #5166
Followed [the instruction](https://github.com/remix-run/react-router/discussions/8753)

What I have done in this PR
- Update `react-router-dom`, `@types/react-router-dom`, and `history` npm libraries
- Replace `react-router-dom-v5-compat` with `react-router-dom`
- Replace `Prompt` cuz its not available in `react-router-dom v6`
  - Fixed bugs [WEB-518]


## Plan

```mermaid
gantt
    title Update React-Router-Dom Plan
    dateFormat  YYYY-MM-DD
    section React-Router v6
    Replace hooks                           :a1, 2022-09-28, 3d
    Replace Root                             :a2, after a1 , 5d
    Wrap up React Router v6         :crit, a3, after a2 , 5d
    section React v18
    Update to React v18      :b1,  2022-10-12, 5d
```




## Test Plan
- Check if all routes works as expected
- Check if prompt shows up if user is still editing `Note` 



## Commentary (optional)
- Lots of files are changed, but mostly renaming `react-router-dom-v5-compat` to `react-router-dom`




## Checklist
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.





[WEB-172]: https://determinedai.atlassian.net/browse/WEB-172?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[WEB-518]: https://determinedai.atlassian.net/browse/WEB-518?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
